src/bodies/polygon.ts

```diff
@@ -163,6 +182,27 @@ export class Polygon extends SATPolygon implements BBox, Collider {
     this.system?.updateBody(this);
   }
 
+  /**
+   * update scale
+   * @param {number} x
+   * @param {number} y
+   */
+  setScale(x: number, y: number = x): void {
+    if (!this.pointsBackup) {
+      this.pointsBackup = clonePointsArray(this.points);
+    }
+
+    this.scaleVector.x = x;
+    this.scaleVector.y = y;
+
+    this.points.forEach((point, i) => {
```

Comment:
```suggestion
    this.points.forEach((point: SATVector, i: number) => {
```

please add types where possible